src/main/java/net/pms/configuration/PmsConfiguration.java

```diff
@@ -2720,12 +2722,29 @@ public void setIpFilter(String value) {
 		configuration.setProperty(KEY_IP_FILTER, value);
 	}
 
-	public void setPreventsSleep(boolean value) {
-		configuration.setProperty(KEY_PREVENTS_SLEEP, value);
-	}
-
-	public boolean isPreventsSleep() {
-		return getBoolean(KEY_PREVENTS_SLEEP, true);
+	public void setPreventSleep(PreventSleepMode value) {
+		if (value == null) {
+			throw new NullPointerException("value cannot be null");
+		}
+		configuration.setProperty(KEY_PREVENT_SLEEP, value.getValue());
+		PMS.get().getSleepManager().setMode(value);
+	}
+
+	public PreventSleepMode getPreventSleep() {
+		PreventSleepMode sleepMode = null;
+		String value = getString(KEY_PREVENT_SLEEP, null);
+		if (value == null && configuration.containsKey(KEY_OLD_PREVENTS_SLEEP)) {
+			// Backwards compatibility
+				sleepMode =
```

Comment:
@SubJunk I was created at some people at Google that wanted something quicker/easier than C. In their quest to make it simple, they left our a whole lot of useful stuff which means the possibility to reuse code is severly limited. Inheritance, overloading, generics are all non-existant for example, basic concepts that's crucial to make "useful" code as I see it. I don't want to keep writing *almost* the same code over and over again.

I like that you can return multiple values, that's a fundamental flaw in Java. You can return multiple objects by returning an array of ```Object```, but then you screw type-safety completely. Even though that's a serious shortcoming, it doesn't outweigh all the fundamental stuff missing from golang IMO. It seems to me that what it's primarily replacing is Python code, and Python isn't a "proper language" to me anyway (although you can do a lot of hackish stuff there easily). 

The one thing I *really* like with golang is that it compiles proper, native binaries without any dependencies. That is what *really* sucks with Java. There does exist such a compiler for Java though, but it's commercial. The day the open source community gets around to writing such a compiler for Java, things will improve a lot. Good bye JVMs, bytecode, JAR files, classpaths and all the rest. Distribute the executable and it can do it's stuff on its own - the only *right* way to do it.